### DISCUSSION TITLE
Add a "button" option for `magit-revision-insert-related-refs`

### DISCUSSION THREAD

**Discussion by vincentbernat** [magit/magit#4735] [Ideas]
Hey!

Inserting related refs is painly slow on some repositories. For example, for Linux repository, it can takes several seconds. This can be fixed by setting `magit-revision-insert-related-refs` to `nil`. However, the information can be quite useful. It could be toggled on demand by allowing to set `'button` for `magit-revision-insert-related-refs`. Upon clicking on the button, the related refs would be computed and the button would be replaced by the result.

**Comment by kyleam** [comment]
[ I realize that 1) you're asking for Magit to support this and 2) you're asking for a button, but in case it's useful to you or others... ]

I've set `magit-revision-insert-related-refs` to nil in my configuration for a long time.  When I want to see the refs, I use this custom command:

```elisp
(defun km/magit-revision-insert-related-refs ()
  (interactive)
  (let ((magit-revision-insert-related-refs t))
    (magit-refresh)))

(define-key magit-revision-mode-map (kbd "C-c C-r")
  #'km/magit-revision-insert-related-refs)
```

**Reply by vincentbernat** [reply]
Thanks, that's pretty useful!
